drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -117,7 +117,14 @@ void stm32_i2c_event_isr(void *arg)
 error:
 	LL_I2C_DisableIT_TX(i2c);
 	LL_I2C_DisableIT_RX(i2c);
-	data->current.is_err = 1;
+
+	if (LL_I2C_IsActiveFlag_NACK(i2c)) {
+		LL_I2C_ClearFlag_NACK(i2c);
+		data->current.is_nack = 1;
+	} else {
+		data->current.is_err = 1;
+	}
```

Comment:
This does not seem to be in the right place in the event handler, it should be handled before looking at the message it self.

Reply:
@vaussard Here is the Linux implementation of the same IP (STM32F7 uses the same i2c IP) https://github.com/torvalds/linux/blob/master/drivers/i2c/busses/i2c-stm32f7.c#L670